### ISSUE TITLE
Laravel 10 support coming soon?

### ISSUE THREAD

**Issue by wallaaa** [ckfinder/ckfinder-laravel-package#92]
Wondering if there are plans to add support for Laravel 10, which was just released earlier today.

It does not appear to require many changes to go from Laravel 9 to 10.

**Comment by bmlotek** [comment]
Hi @wallaaa, we have this update in our roadmap. The update might not require many changes on the first sight, but some libraries are shared between CKFinder and Laravel which sometimes can cause conflicts. We need to make sure the integration works as expected with the newest version.

**Comment by theupriser** [comment]
@wallaaa @bmlotek  I've just forked the repo and put the dependencies in the composer.json for Laravel 10. Everything works as expected. Upgrading the package literally takes a minute and a half.
Laravel 10 does not change anything with flysystem and in my eyes it's just a minor upgrade.

**Comment by bmlotek** [comment]
@theupriser hello, thank you for you work. @wallaaa CKFinder Laravel Package v4.0.1 with support for Laravel 10 just has been released. Enjoy.

https://packagist.org/packages/ckfinder/ckfinder-laravel-package

**Comment by olivierlacan** [comment]
@bmlotek You can probably close this issue although I'll note that `composer require ckfinder/ckfinder-laravel-package` fails with a fresh Laravel 10 install unless you pass the `--with-all-dependencies` or `-W` flag as well due to a version conflict.

See below for output (without `-W` flag): 
```
Info from https://repo.packagist.org: #StandWithUkraine
./composer.json has been updated
Running composer update ckfinder/ckfinder-laravel-package
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - league/flysystem-aws-s3-v3[3.12.1, ..., 3.15.0] require aws/aws-sdk-php ^3.220.0 -> satisfiable by aws/aws-sdk-php[3.220.0, ..., 3.275.2].
    - league/flysystem-aws-s3-v3[3.0.0, ..., 3.10.3] require aws/aws-sdk-php ^3.132.4 -> satisfiable by aws/aws-sdk-php[3.132.4, ..., 3.275.2].
    - ckfinder/ckfinder-laravel-package[v3.4.4, ..., v3.5.1] require laravel/framework ^5.5 -> found laravel/framework[v5.5.0, ..., v5.8.38] but it conflicts with your root composer.json require (^10.10).
    - ckfinder/ckfinder-laravel-package[v3.5.1.1, ..., v3.5.1.2] require laravel/framework ^5.5|~6.0 -> found laravel/framework[v5.5.0, ..., v5.8.38, v6.0.0, ..., v6.20.44] but it conflicts with your root composer.json require (^10.10).
    - ckfinder/ckfinder-laravel-package[v3.5.1.3, ..., v3.5.1.5] require laravel/framework ^5.5|~6.0|~7.0 -> found laravel/framework[v5.5.0, ..., v5.8.38, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6] but it conflicts with your root composer.json require (^10.10).
    - ckfinder/ckfinder-laravel-package[v3.5.1.6, ..., v3.5.2.1] require laravel/framework ^5.5|^6.0|^7.0|^8.0 -> found laravel/framework[v5.5.0, ..., v5.8.38, v6.0.0, ..., v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but it conflicts with your root composer.json require (^10.10).
    - ckfinder/ckfinder-laravel-package v3.5.3 requires laravel/framework ^5.5|^6.20.42|^7.0|^8.0 -> found laravel/framework[v5.5.0, ..., v5.8.38, v6.20.42, v6.20.43, v6.20.44, v7.0.0, ..., v7.30.6, v8.0.0, ..., v8.83.27] but it conflicts with your root composer.json require (^10.10).
    - ckfinder/ckfinder-laravel-package v4.0.0 requires laravel/framework ^9.0 -> found laravel/framework[v9.0.0, ..., v9.52.10] but it conflicts with your root composer.json require (^10.10).
    - aws/aws-sdk-php[3.131.0, ..., 3.173.19] require guzzlehttp/psr7 ^1.4.1 -> found guzzlehttp/psr7[1.4.1, ..., 1.9.1] but the package is fixed to 2.5.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - aws/aws-sdk-php[3.173.20, ..., 3.198.8] require guzzlehttp/psr7 ^1.7.0 -> found guzzlehttp/psr7[1.7.0, ..., 1.9.1] but the package is fixed to 2.5.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - aws/aws-sdk-php[3.199.0, ..., 3.275.2] require guzzlehttp/promises ^1.4.0 -> found guzzlehttp/promises[1.4.0, ..., 1.5.3] but the package is fixed to 2.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - ckfinder/ckfinder-laravel-package v4.0.1 requires league/flysystem-aws-s3-v3 ^3.0 -> satisfiable by league/flysystem-aws-s3-v3[3.0.0, ..., 3.15.0].
    - Root composer.json requires ckfinder/ckfinder-laravel-package * -> satisfiable by ckfinder/ckfinder-laravel-package[v3.4.4, ..., v3.5.3, v4.0.0, v4.0.1].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require ckfinder/ckfinder-laravel-package:*" to figure out if any version is installable, or "composer require ckfinder/ckfinder-laravel-package:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

**Comment by bmlotek** [comment]
Hello @olivierlacan thank you for heads up. I had same error on newest (v10.2.4) version of Laravel. I will check it out as soon as possible.

**Comment by bmlotek** [comment]
@olivierlacan sorry for such a late answer. 

From my brief analysis i have notice that:

There is problem with this package `package league/flysystem-aws-s3-v3`, because if you run (on fresh Laravel instance) `composer require league/flysystem-aws-s3-v3` without `--with-all-dependencies` it fails with similar errors:

```
Using version ^3.15 for league/flysystem-aws-s3-v3
./composer.json has been updated
Running composer update league/flysystem-aws-s3-v3
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - aws/aws-sdk-php[3.220.0, ..., 3.278.3] require guzzlehttp/promises ^1.4.0 -> found guzzlehttp/promises[1.4.0, ..., 1.5.3] but the package is fixed to 2.0.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - league/flysystem-aws-s3-v3 3.15.0 requires aws/aws-sdk-php ^3.220.0 -> satisfiable by aws/aws-sdk-php[3.220.0, ..., 3.278.3].
    - Root composer.json requires league/flysystem-aws-s3-v3 ^3.15 -> satisfiable by league/flysystem-aws-s3-v3[3.15.0].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require league/flysystem-aws-s3-v3:*" to figure out if any version is installable, or "composer require league/flysystem-aws-s3-v3:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

But if you run `composer require league/flysystem-aws-s3-v3` with `-W` no errors appear and after that you can run `composer require ckfinder/ckfinder-laravel-package` also without error. So as you already wrote before, just run `composer require ckfinder/ckfinder-laravel-package` with `--with-all-dependencies` or `-W` option.
